atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -220,6 +220,8 @@
     private final Timer.Context transactionTimerContext;
     protected final CommitProfileProcessor commitProfileProcessor;
 
+    protected final TransactionOutcomeMetrics transactionOutcomeMetrics;
```

Comment:
nit: no reason why this field and the two above it can't belong in the main block of (mostly) `final` fields.

Reply:
Fair enough - indeed it looks weird that the volatile boolean is randomly sitting in the middle of final fields. Moved them up.